pkg/selector/aws/selector.go

```diff
@@ -17,16 +17,149 @@ package aws
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awscfg "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
 	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/config"
+	"github.com/chaos-mesh/chaos-mesh/pkg/mock"
+	"github.com/chaos-mesh/chaos-mesh/pkg/selector/generic"
+	"go.uber.org/fx"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type SelectImpl struct{}
+// EC2Client defines the minimum client interface required for this package
+type EC2Client interface {
+	DescribeInstances(context.Context, *ec2.DescribeInstancesInput, ...func(*ec2.Options)) (*ec2.DescribeInstancesOutput, error)
+}
+
+type SelectImpl struct {
+	c client.Client
+	generic.Option
+}
 
 func (impl *SelectImpl) Select(ctx context.Context, awsSelector *v1alpha1.AWSSelector) ([]*v1alpha1.AWSSelector, error) {
```

Comment:
It seems that we could use an other type instead of `[]*v1alpha1.AWSSelector` in the return value as the "selected records". `v1alpha1.AWSSelector` carries too much noise info that we do not need in the next steps.

How about creating a new struct to introduce the selected ec2 or volume info?

also cc @cwen0  @YangKeao

Reply:
Yes, I agree it's better.  I added `aws.Instance` struct type for this.  Thanks!